### ISSUE TITLE
What exact MC7455 firmware version can be unlocked

### ISSUE THREAD

**Issue by amore-amore** [bkerler/SierraWirelessGen#2]
using the calc?

**Comment by amore-amore** [comment]
I've tested all the available firmware for MC7455 to unlock but with no luck.

swi9x30c_02.08.02.00.cwe swi9x30c_02.08.02.00_generic_002.007_000.nvu
swi9x30c_02.14.03.00.cwe swi9x30c_02.14.03.00_generic_002.012_000.nvu
swi9x30c_02.18.02.00.cwe swi9x30c_02.18.02.00_generic_002.015_000.nvu
swi9x30c_02.20.03.00.cwe swi9x30c_02.20.03.00_generic_002.017_000.nvu
swi9x30c_02.23.00.00.cwe swi9x30c_02.23.00.00_generic_002.018_000.nvu

Could you write an exact FW version where it works?

**Comment by bkerler** [comment]
The reverse engineered firmware was SWI9X30C_02.24.05.06.cwe

**Comment by panachoi** [comment]
> The reverse engineered firmware was SWI9X30C_02.24.05.06.cwe

Does this work for EM series (M.2) ? I have a lenovo-branded EM7455 running this exact firmware, and cannot get it to unlock.

**Comment by schoerg** [comment]
Try it. Shouldn't matter if it's m.2 or PCIe.

**Comment by panachoi** [comment]
Tried just about every combination available, and could not get it to unlock with any of the algos.

**Comment by bkerler** [comment]
The main issue is that most of these devices have encrypted firmware that cannot be decrypted without known AES keys. Thus reversing is only possible if the firmware is unencrypted.

**Comment by gsmantenna** [comment]
I downgrade the firmware to SWI9X30C_02.24.05.06, and try from V2-V5 openlock code, none work, what am I missing?

**Comment by bkerler** [comment]
I had a look at the newer firmwares and they use a different scheme. Can you please send me an example output of "AT!OPENLOCK?" and "AT!OPENMEP?" ?

**Comment by bkerler** [comment]
Also please state which modem vendor and maybe provider (if you live in the US) you use (like dell, provider at&t)

**Comment by schoerg** [comment]
```
AT!OPENLOCK?
1B7A923CCC5AE55E

AT!OPENLOCK?
0621C5308B38F615

AT!OPENMEP?
E04D3018BF64AB5A

AT!OPENMEP?
805355F8C308F35F
```

Device:
```
Manufacturer: Sierra Wireless, Incorporated
Model: EM7455
Revision: SWI9X30C_02.20.03.00 r6691 CARMD-EV-FRMWR2 2016/06/30 10:54:05

```

**Comment by bkerler** [comment]
Thx a lot.

**Comment by gsmantenna** [comment]
Model: EM7455B
Revision: SWI9X30C_02.24.05.06 r7040 CARMD-EV-FRMWR2 2017/05/19 06:23:09
MEID: 354480080xxxxx
IMEI: 3544800800xxxxx
IMEI SV: 12
FSN: LF637643460210
+GCAP: +CGSM


OK
AT!OPENLOCK?
4E3DDE95D157D4DB

OK
AT!OPENLOCK?
8A30B8401EA00320

OK
AT!OPENMEP?
F480C6769B7A00F3

OK

AT!OPENMEP?
BE9F0A5AF544CF50
dell em7455 un-branded.

**Comment by bkerler** [comment]
Solved by commit 6cade1d
